### ISSUE TITLE
GSSAPI Extended Test Failure

### ISSUE THREAD

**Issue by enj** [openshift/origin#11013]
For some reason `oc whoami` is returning `system:serviceaccount:gssapiproxy:default` instead of `system:anonymous`.

##### Version
```
openshift v1.4.0-alpha.0+85eb37b
kubernetes v1.4.0-beta.3+d19513f
etcd 3.0.6

oc v1.3.0-rc1+e9081ae
kubernetes v1.3.0+52492b4
features: Basic-Auth GSSAPI Kerberos SPNEGO
```

##### Steps To Reproduce
1. Fedora base image, client has uncofigured GSSAPI libs and server is GSSAPI only
2. `oc whoami`

##### Current Result
`system:serviceaccount:gssapiproxy:default`

##### Expected Result
`system:anonymous`

##### Additional Information
https://ci.openshift.redhat.com/jenkins/job/gssapi_extended/52/

```
Regression

github.com/openshift/origin/test/test-extended/gssapiproxy-tests/fedora-gssapi-base-CLIENT_HAS_LIBS-SERVER_GSSAPI_ONLY../gssapi-tests.sh:62: executing 'oc whoami' expecting failure and text 'system:anonymous' (from github.com_openshift_origin_test_test-extended_gssapiproxy-tests_fedora-gssapi-base-CLIENT_HAS_LIBS-SERVER_GSSAPI_ONLY)
Failing for the past 1 build (Since Failed#52 )
Took 0.21 sec.
Error Message

the command returned the wrong error code; the output content test failed

Stacktrace

=== BEGIN TEST CASE ===
./gssapi-tests.sh:62: executing 'oc whoami' expecting failure and text 'system:anonymous'
FAILURE after 0.219s: ./gssapi-tests.sh:62: executing 'oc whoami' expecting failure and text 'system:anonymous': the command returned the wrong error code; the output content test failed
Standard output from the command:
system:serviceaccount:gssapiproxy:default

There was no error output from the command.
=== END TEST CASE ===
```

cc @liggitt @smarterclayton (started on Sept 16 so may be related to the rebase)

**Comment by liggitt** [comment]
I think the pod is picking up the mounted token for the service account more effectively. Mount an emptydir at /var/run/secrets/kubernetes.io/serviceaccount to prevent picking up the token inside the pod.

**Comment by liggitt** [comment]
@smarterclayton did your changes make the incluster config get used even when KUBECONFIG was pointing to an empty/anonymous config?

**Comment by enj** [comment]
For reference, the `KUBECONFIG` used in the test is:
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    server: https://openshift.default.svc.cluster.local:443
  name: GSSAPITEST
contexts:
- context:
    cluster: GSSAPITEST
    namespace: gssapiproxy
  name: gssapi
current-context: gssapi
kind: Config
preferences: {}
users: []
```

**Comment by liggitt** [comment]
If given a kubeconfig like that, we shouldn't be falling back to incluster config (picking up the service account token from disk)

**Comment by liggitt** [comment]
@smarterclayton is this related to the upstream issue with inclusterconfig reading?

**Comment by enj** [comment]
I believe https://github.com/kubernetes/kubernetes/pull/33136 fixes this, will do some tests and report back.

**Comment by enj** [comment]
@liggitt @smarterclayton https://github.com/kubernetes/kubernetes/pull/33136 does fix this issue.  I am not sure what the process for cherry picking is.

**Comment by enj** [comment]
Fixed by #11115
